src/commands/CodingJamsCommand.ts

```diff
@@ -0,0 +1,307 @@
+import { Command } from "../interactions/interactionClasses";
+import {
+	ChatInputCommandInteraction,
+	Client,
+	SlashCommandBuilder,
+	SlashCommandIntegerOption,
+	SlashCommandStringOption,
+	SlashCommandSubcommandsOnlyBuilder,
+} from "discord.js";
+import { hasAdminPerms } from "../util/misc/permissions";
+import { deleteJam, editJam, newJam, viewJam } from "../util/coding-jams/manageJams";
+import { deletePoll, editPoll, newPoll, viewPoll, votesPoll } from "../util/coding-jams/managePolls";
+import { jamSchedulerTick } from "../util/coding-jams/eventHandler";
+import cron from "node-cron";
+import { checkDate, checkDuration } from "../util/misc/time";
+import { DateTime, Duration } from "luxon";
+import { listJams, listPolls } from "../util/coding-jams/listPollsAndJams";
+
+class CodingJamsCommand extends Command {
+	constructor() {
+		super("coding-jams");
+	}
+
+	startScheduler(client: Client) {
+		jamSchedulerTick(client);
+		// every 30 min
+		cron.schedule("*/30 * * * *", jamSchedulerTick.bind(this, client));
+	}
+
+	async execute(interaction: ChatInputCommandInteraction): Promise<void> {
+		if (!(await hasAdminPerms(interaction))) {
+			await interaction.reply({
+				content: "The coding jam commands are meant to be used by admins only.",
+				ephemeral: true,
+			});
+			return;
+		}
+
+		const name = interaction.options.getString("name") || "Default Name";
+		const proposal = interaction.options.getString("proposal") || "";
+
+		const numVotes = interaction.options.getInteger("num-votes") || 3;
+		const numProposals = interaction.options.getInteger("num-proposals") || 3;
+		const selectionType =
+			interaction.options.getString("proposal-selection-type") || pollSelectionTypes.topAll;
+
+		const subCmdGroup = interaction.options.getSubcommandGroup() || "cmd";
+		const subCmd = interaction.options.getSubcommand();
+
+		let startDate: DateTime | null;
+		let endDate: DateTime | null;
+		let duration: Duration | null;
+
+		if (subCmd === "new") {
+			startDate = await checkDate(interaction, interaction.options.getString("start-date"));
+			endDate = await checkDate(interaction, interaction.options.getString("end-date"));
+			duration = await checkDuration(interaction, interaction.options.getString("duration"));
+
+			if (!startDate) return;
```

Comment:
Shouldn't there be a error message?

Reply:
No. Is already in `checkDate()` included